### PR TITLE
Attempt to fix the v15 release line to not be backward-compat breaking

### PR DESCRIPTION
**Pull request by domenic**
This attempts to revert all changes in master plus v15.2.0 that were backward-compat breaking, so that we can release a v15.2.1 that backward-compatible with v15.1.0 and v15.0.0. Such a v15.2.1 would have the following deltas from v15.1.0:

- Full support for callback interfaces
- Updated webidl-conversions dependency

(The addition of `new()` is reverted for simplicity, since it was based on the removal of `init()`.)

It would be nice if we could unpublish v15.2.0, but apparently we can't since it has dependent packages in the registry.

The intention would then be to un-revert these four commits (except maybe not the removal of `init()`; see https://github.com/jsdom/webidl2js/pull/195#pullrequestreview-395813297) and publish a v16.0.0 with the breaking changes of requiring `[Exposed]` on all interfaces and requiring a second argument to `install()`.

This is the best I can currently think of, but I welcome a second pair of eyes, since I already messed this up.

### CODE REVIEW

File: lib/constructs/callback-interface.js [hunk 1]
```diff
@@ -197,24 +178,14 @@ class CallbackInterface {
...
   generateInstall() {
-    if (this.constants.size > 0) {
-      this.str += `
-        const exposed = new Set(${JSON.stringify([...this.exposed])});
-      `;
-    }
-
     this.str += `
-      exports.install = (globalObject, globalNames) => {
+      exports.install = function install(globalObject) {
```
Comment (ExE-Boss) [inline lib/constructs/callback-interface.js:182]:
```suggestion
      exports.install = (globalObject, globalNames = ["Window"]) => {
```

File: test/cases/CEReactions.webidl [hunk 2]
```diff
@@ -1,8 +1,7 @@
-[Exposed=Window]
 interface CEReactions {
     [CEReactions] attribute DOMString attr;
     [CEReactions] void method();
-
+    
     getter DOMString (DOMString name);
```
Comment (ExE-Boss) [inline test/cases/CEReactions.webidl:5]:
```suggestion

    getter DOMString (DOMString name);
```

File: test/cases/CEReactions.webidl [hunk 3]
```diff
@@ -1,8 +1,7 @@
-[Exposed=Window]
 interface CEReactions {
```
Comment (ExE-Boss) [inline test/cases/CEReactions.webidl:1]:
```suggestion
[Exposed=Window]
interface CEReactions {
```

File: test/cases/Global.webidl [hunk 4]
```diff
@@ -1,4 +1,4 @@
-[Global=Global,Exposed=Global]
+[Global=Global,Exposed=Window]
```
Comment (ExE-Boss) [inline test/cases/Global.webidl:1]:
```suggestion
[Global=Global,Exposed=Global]
```

File: test/cases/HTMLConstructor.webidl [hunk 5]
```diff
@@ -1,3 +1,2 @@
-[Exposed=Window,
- HTMLConstructor]
+[HTMLConstructor]
```
Comment (ExE-Boss) [inline test/cases/HTMLConstructor.webidl:1]:
```suggestion
[Exposed=Window,
 HTMLConstructor]
```

File: test/cases/URL.webidl [hunk 6]
```diff
@@ -1,5 +1,4 @@
-[Exposed=(Window,Worker),
- LegacyWindowAlias=webkitURL]
+[Exposed=(Window,Worker)]
```
Comment (ExE-Boss) [inline test/cases/URL.webidl:1]:
```suggestion
[Exposed=(Window,Worker),
 LegacyWindowAlias=webkitURL]
```

**Comment by ExE-Boss** [comment]
Alternatively, just create a new branch `v15` and backport things to it.

**Comment by domenic** [comment]
Hey @ExE-Boss, please refrain from reviewing in the future; your review was not requested and it's not polite to do drive-by reviews in this way.

Making globalNames optional will not solve the backward-compat issue that [Exposed] is now required, I believe.

**Comment by domenic** [comment]
Replaced by https://github.com/jsdom/webidl2js/pull/202.

**Pull request closed without merging**